test/test1979.js

```diff
@@ -0,0 +1,135 @@
+			alasql('SELECT NOW() AS NOW');
+		});
+		assert.throws(function() {
+			alasql('SELECT NOW() AS NOW()');
+		});
+	});
+
+	it('1b. GETDATE is a reserved keyword', function () {
+		assert.throws(function() {
+			alasql('SELECT GETDATE() AS GETDATE');
+		});
+		assert.throws(function() {
+			alasql('SELECT GETDATE() AS getdate');
+		});
+		assert.throws(function() {
+			alasql('SELECT GETDATE() AS getdate()');
+		});
+	});
+
+	it('1c. Check parser', function () {
+		// If the parser works ok, the date functions are parsed with a funcid instead of columnid
+		assert.equal("CURRENT_TIMESTAMP", alasql.parse('SELECT CURRENT_TIMESTAMP').statements[0].columns[0].funcid);
+		assert.equal("CURRENT_TIMESTAMP", alasql.parse('SELECT CURRENT_TIMESTAMP()').statements[0].columns[0].funcid);
+		assert.equal("CURRENT_TIMESTAMP", alasql.parse('SELECT NOW').statements[0].columns[0].funcid);
+		assert.equal("NOW", alasql.parse('SELECT NOW()').statements[0].columns[0].funcid);
+		assert.equal("CURRENT_TIMESTAMP", alasql.parse('SELECT GETDATE').statements[0].columns[0].funcid);
+		assert.equal("GETDATE", alasql.parse('SELECT GETDATE()').statements[0].columns[0].funcid);
+		assert.equal("CURRENT_DATE", alasql.parse('SELECT CURRENT_DATE').statements[0].columns[0].funcid);
+		assert.equal("CURRENT_DATE", alasql.parse('SELECT CURRENT_DATE()').statements[0].columns[0].funcid);
+		assert.equal("CURRENT_DATE", alasql.parse('SELECT CURDATE').statements[0].columns[0].funcid);
+		assert.equal("CURDATE", alasql.parse('SELECT CURDATE()').statements[0].columns[0].funcid);
+
+		assert.equal("SECOND", alasql.parse('SELECT SECOND(NOW())').statements[0].columns[0].funcid);
+		assert.equal("HOUR", alasql.parse('SELECT HOUR(NOW())').statements[0].columns[0].funcid);
+		assert.equal("MINUTE", alasql.parse('SELECT MINUTE(NOW())').statements[0].columns[0].funcid);
+		assert.equal("DAY", alasql.parse('SELECT DAY(NOW())').statements[0].columns[0].funcid);
+		assert.equal("DAYOFWEEK", alasql.parse('SELECT DAYOFWEEK(NOW())').statements[0].columns[0].funcid);
+		assert.equal("MONTH", alasql.parse('SELECT MONTH(NOW())').statements[0].columns[0].funcid);
+		assert.equal("YEAR", alasql.parse('SELECT YEAR(NOW())').statements[0].columns[0].funcid);
+
+		assert.equal("DATE_ADD", alasql.parse('SELECT DATE_ADD(DATE("20081012"), INTERVAL 10 DAY)').statements[0].columns[0].funcid);
+		assert.equal("DATE_ADD", alasql.parse('SELECT ADDDATE(DATE("20081012"), INTERVAL 10 DAY)').statements[0].columns[0].funcid);
+		assert.equal("DATE_SUB", alasql.parse('SELECT DATE_SUB(DATE("20081012"), INTERVAL 10 DAY)').statements[0].columns[0].funcid);
+		assert.equal("DATE_SUB", alasql.parse('SELECT SUBDATE(DATE("20081012"), INTERVAL 10 DAY)').statements[0].columns[0].funcid);
+		assert.equal("DATE", alasql.parse('SELECT DATE("20081012")').statements[0].columns[0].funcid);
+	});
+
+	it('1d. SECOND/HOUR/MINUTE/DAY/DAYOFWEEK/MONTH/YEAR are reserved keywords', function () {
+		assert.throws(function() {
```

Comment:
FYI this is the section of tests im working on making sure this does not throws. so we dont break too many things.